api/api.go

```diff
@@ -35,120 +30,115 @@ func HttpPOST(url string, data string) (resp *http.Response, err error) {
-		srv.handleHTTPRequest(mux, "/consensus/block", srv.consensusBlockHandler) // GET
+		mux.GET("/consensus", srv.consensusHandler)
+		mux.GET("/consensus/blocks/:height", srv.consensusBlocksHandler)
 	}
 
 	// Explorer API Calls
 	if srv.explorer != nil {
-		srv.handleHTTPRequest(mux, "/explorer", srv.explorerHandler)            // GET
-		srv.handleHTTPRequest(mux, "/explorer/", srv.explorerHandler)           // $(hash) GET
-		srv.handleHTTPRequest(mux, "/explorer/block", srv.explorerBlockHandler) // GET
+		mux.GET("/explorer", srv.explorerHandler)
+		mux.GET("/explorer/blocks/:height", srv.explorerBlocksHandler)
+		mux.GET("/explorer/hash/:hash", srv.explorerHashHandler)
 	}
 
 	// Gateway API Calls - Unfinished
 	if srv.gateway != nil {
-		srv.handleHTTPRequest(mux, "/gateway/status", srv.gatewayStatusHandler)
-		srv.handleHTTPRequest(mux, "/gateway/peers/add", srv.gatewayPeersAddHandler)
-		srv.handleHTTPRequest(mux, "/gateway/peers/remove", srv.gatewayPeersRemoveHandler)
+		mux.GET("/gateway", srv.gatewayHandler)
+		mux.POST("/gateway/add/:addr", srv.gatewayAddHandler)
+		mux.POST("/gateway/remove/:addr", srv.gatewayRemoveHandler)
 	}
 
 	// Host API Calls
 	if srv.host != nil {
-		srv.handleHTTPRequest(mux, "/host", srv.hostHandler)                  // GET, POST
-		srv.handleHTTPRequest(mux, "/host/announce", srv.hostAnnounceHandler) // POST
+		mux.GET("/host", srv.hostHandlerGET)
+		mux.POST("/host", srv.hostHandlerPOST)
+		mux.POST("/host/announce", srv.hostAnnounceHandler)
 	}
 
 	// HostDB API Calls - DEPRECATED
 	if srv.renter != nil {
-		srv.handleHTTPRequest(mux, "/hostdb/hosts/active", srv.renterHostsActiveHandler)
-		srv.handleHTTPRequest(mux, "/hostdb/hosts/all", srv.renterHostsAllHandler)
+		mux.HandlerFunc("GET", "/hostdb/hosts/active", srv.renterHostsActiveHandler)
+		mux.HandlerFunc("GET", "/hostdb/hosts/all", srv.renterHostsAllHandler)
 	}
 
 	// Miner API Calls
 	if srv.miner != nil {
-		srv.handleHTTPRequest(mux, "/miner", srv.minerHandler)                            // GET
-		srv.handleHTTPRequest(mux, "/miner/header", srv.minerHeaderHandler)               // GET, POST
-		srv.handleHTTPRequest(mux, "/miner/start", srv.minerStartHandler)                 // POST
-		srv.handleHTTPRequest(mux, "/miner/stop", srv.minerStopHandler)                   // POST
-		srv.handleHTTPRequest(mux, "/miner/headerforwork", srv.minerHeaderforworkHandler) // COMPATv0.4.8
-		srv.handleHTTPRequest(mux, "/miner/submitheader", srv.minerSubmitheaderHandler)   // COMPATv0.4.8
+		mux.GET("/miner", srv.minerHandler)
+		mux.GET("/miner/header", srv.minerHeaderHandlerGET)
+		mux.POST("/miner/header", srv.minerHeaderHandlerPOST)
+		mux.POST("/miner/start", srv.minerStartHandler)
+		mux.POST("/miner/stop", srv.minerStopHandler)
+		mux.GET("/miner/headerforwork", srv.minerHeaderHandlerGET) // COMPATv0.4.8
+		mux.GET("/miner/submitheader", srv.minerHeaderHandlerPOST) // COMPATv0.4.8
```

Comment:
These do need to be preserved though because a huge part of the hashrate depends on these calls.